_analyzers/token-filters/keyword-marker.md

```diff
@@ -0,0 +1,169 @@
+---
+layout: default
+title: Keyword marker
+parent: Token filters
+nav_order: 200
+---
+
+# Keyword marker token filter
+
+The `keyword_marker` token filter in OpenSearch is used to prevent certain tokens from being altered by stemmers or other filters. This ensures that specific words remain in their original form. `keyword_marker` token filter does this by marking tokens as `keywords` which prevents any stemming or other processing.
+
+## Parameters
+
+The `keyword_marker` token filter in OpenSearch can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`ignore_case` | Optional | Boolean | Ignore the letter case when matching keywords. Default is `false.`
+`keywords` | Required if `keywords_path` or `keywords_pattern` is not set | List of strings | List of strings used to match tokens.
+`keywords_path` | Required if `keywords` or `keywords_pattern` is not set | String | Path (relative to `config` directory or absolute) to list of strings.
+`keywords_pattern` | Required if `keywords` or `keywords_path` is not set | String | [Regular expression](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) used for matching tokens.
+ 
+
+## Example
+
+The following example request creates a new index named `my_index` and configures an analyzer with `keyword_marker` filter:
```

Comment:
Not sure if OpenSearch is the best word for showcasing the filter. I would shorten the sentence to a couple of words which are normally stemmed. Then show examples with and without the filter so users can see that the filter prevents stemming.